extra-kotlin/src/main/kotlin/net/kyori/adventure/extra/kotlin/StyleExtensions.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2021 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.extra.kotlin
+
+import net.kyori.adventure.text.format.Style
+import net.kyori.adventure.text.format.TextDecoration
+import org.jetbrains.annotations.Contract
+
+/**
+ * Check if this style contains the specified [decoration]
+ *
+ * @return true if this style contains the decoration, false otherwise
+ * @since 4.8.0
+ */
+public operator fun Style.contains(decoration: TextDecoration): Boolean = hasDecoration(decoration)
```

Comment:
Not sure I like these -- since they just operate on TextDecorations rather than any attribute on a style. Can we keep just the `edit` method?

Reply:
Guess we could? Though the purpose of the `operator` functions here is that they're much more idiomatic. The `set` functions don't really make that much sense I suppose, since that's designed for things like setting the value of a key in a mutable map, but `contains` should stay imo